src/pip/_internal/network/session.py

```diff
@@ -238,6 +238,9 @@ def __init__(self, *args, **kwargs):
         cache = kwargs.pop("cache", None)
         trusted_hosts = kwargs.pop("trusted_hosts", [])  # type: List[str]
         index_urls = kwargs.pop("index_urls", None)
+        auth_class = kwargs.pop("auth_class", None)
+        if auth_class is None:
+            auth_class = MultiDomainBasicAuth
```

Comment:
```suggestion
        auth_class = kwargs.pop("auth_class",  MultiDomainBasicAuth)
```

Reply:
That does not work because this named argument is always provided.

Reply:
Thanks, I missed that.  How about provide it with `MultiDomainBasicAuth` as suggested below?

Reply:
I tried that at first. But None is a better default IMO. This way cli is not coupled with network. Currently MultiDomainBasicAuth is not a plugin, it's the default implementation.